### ISSUE TITLE
WAYLAND_DISPLAY is not set, defaulting to "wayland-0"

### ISSUE THREAD

**Issue by Mamoga** [waydroid/waydroid#771]
**Describe the bug**
[15:45:37] WAYLAND_DISPLAY is not set, defaulting to "wayland-0"
[15:45:37] Wayland socket '/run/user/1000/wayland-0' doesn't exist; are you running a Wayland compositor?

 - Waydroid tools Version [latest]
 - Waydroid Images Version [VANILLA] 
-  OS: [Ubuntu 22.04 x64]
 - GPU: [MX150]

**Comment by MatthewYounkins** [comment]
I've been trying to fix this issue as well for the past few days, on Ubuntu 22.10.  I followed the instructions at https://docs.waydro.id/usage/install-on-desktops , uninstalled and reinstalled, and still the same error.

After install, I click on the Waydroid icon in the applications menu, and there's no (easily?) observable response.   I try 'waydroid session start' at the terminal and get the same error the OP gets.

**Comment by HanoJing** [comment]
What Desktop environment do you use?
Make sure that XDG_SESSION_TYPE is set to wayland:
`export XDG_SESSION_TYPE=wayland`

you can try to modify the `Exec` lines of `~/.local/share/applications/Waydroid.desktop` and `/usr/share/applications/Waydroid.desktop`  like so: `env WAYLAND_DISPLAY=wayland-0 waydroid show-full-ui` or  `env WAYLAND_DISPLAY=wayland-0 waydroid first-launch`.

**Comment by MatthewYounkins** [comment]
For me, there is no Waydroid.desktop file in ~/.local/share/applications.  I didn't try to copy it over.

For /usr/share, Before:  

![image](https://user-images.githubusercontent.com/26656940/220495937-c228db01-5a71-4ca5-af73-cca12be15065.png)

After:

![image](https://user-images.githubusercontent.com/26656940/220496337-4cbc8adb-13b8-4fcd-9a91-3818129e8ed9.png)

And I still get the same error message.

**Comment by iamharshit188** [comment]
> For me, there is no Waydroid.desktop file in ~/.local/share/applications. I didn't try to copy it over.
> 
> For /usr/share, Before:
> 
> ![image](https://user-images.githubusercontent.com/26656940/220495937-c228db01-5a71-4ca5-af73-cca12be15065.png)
> 
> After:
> 
> ![image](https://user-images.githubusercontent.com/26656940/220496337-4cbc8adb-13b8-4fcd-9a91-3818129e8ed9.png)
> 
> And I still get the same error message.

show me your $XDG_SESSION_TYPE
also in which desktop environment are you executing it ?

**Comment by MatthewYounkins** [comment]
> show me your **$XDG_SESSION_TYPE** echo $XDG_SESSION_TYPE
> 
> also in which desktop environment are you executing it ?

echo $XDG_SESSION_TYPE returns x11

My desktop environment is gnome.  Thoughts?

**Comment by HanoJing** [comment]
> > show me your **$XDG_SESSION_TYPE** echo $XDG_SESSION_TYPE
> > also in which desktop environment are you executing it ?
> 
> echo $XDG_SESSION_TYPE returns x11
> 
> My desktop environment is gnome. Thoughts?

Ah, I guess that's why then. 
If you use a login manager, please logout and set the Destkop Environment to GNOME Wayland in the bottom right corner of your login screen (cog icon).

**Comment by MatthewYounkins** [comment]
> If you use a login manager, please logout and set the Destkop Environment to GNOME Wayland in the bottom right corner of your login screen (cog icon).

I don't use a login manager, I boot straight into my account.  Is a login manager required?

**Comment by HanoJing** [comment]
> > If you use a login manager, please logout and set the Destkop Environment to GNOME Wayland in the bottom right corner of your login screen (cog icon).
> 
> I don't use a login manager, I boot straight into my account. Is a login manager required?

no, it's not required, but for the ease of use, with a login manager you can set your desktop environment for example, it will show GNOME and GNOME Xorg. "GNOME" will be GNOME under wayland.

**Comment by Mamoga** [comment]
> > Если вы используете диспетчер входа в систему, выйдите из системы и установите для среды Destkop значение GNOME Wayland в правом нижнем углу экрана входа (значок шестеренки).
> 
> Я не использую менеджер входа в систему, я загружаюсь прямо в свою учетную запись. Требуется ли менеджер входа?

Try this 

`sudo nano /etc/gdm3/custom.conf`

`WaylandEnable=true` and remove `#`

reboot 

chose Ubuntu on Wayland like this https://www.andreyolegovich.ru/linux/ubuntu/wayland/?ysclid=lej9iyh0vp693670635#turnoff

It's work for me

**Comment by iamharshit188** [comment]
> 

sir , u need to use wayland in order to use waydroid , either use gnome wayland or plasma wayland

**Comment by Mamoga** [comment]
@aleasto can you close close this as completed, please

**Comment by UGoingNoWhereBoy** [comment]
> > > If you use a login manager, please logout and set the Destkop Environment to GNOME Wayland in the bottom right corner of your login screen (cog icon).
> > 
> > 
> > I don't use a login manager, I boot straight into my account. Is a login manager required?
> 
> No, it's not required, but for the ease of use, with a login manager you can set your desktop environment for example, it will show GNOME and GNOME Xorg. "GNOME" will be GNOME under wayland. I found a good guide that shows you how to select wayland: https://linuxconfig.org/how-to-enable-disable-wayland-on-ubuntu-22-04-desktop Also be sure to have `gnome-session-wayland` installed.

Thanks, this worked for me.

**Comment by bobwatcherx** [comment]
how to use in awesome WM
i get error 
mino@mino-HP-Convertible-x360-11-ab1XX:~$ waydroid session start
[10:46:33] WAYLAND_DISPLAY is not set, defaulting to "wayland-0"
[10:46:33] Wayland socket '/run/user/1000/wayland-0' doesn't exist; are you running a Wayland compositor?

**Comment by HanoJing** [comment]
> @aleasto can you close close this as completed, please

You can close this yourself as completed, since you are the author of the issue.

click the button "Close issue" with the purple circle down at the bottom.

**Comment by HanoJing** [comment]
> how to use in awesome WM iam using Linux MINT os i get error mino@mino-HP-Convertible-x360-11-ab1XX:~$ waydroid session start [10:46:33] WAYLAND_DISPLAY is not set, defaulting to "wayland-0" [10:46:33] Wayland socket '/run/user/1000/wayland-0' doesn't exist; are you running a Wayland compositor?

If you want to stay on awesome, you can install weston, start weston, open up a terminal inside of weston and start waydroid from within weston.

`sudo apt install weston`
execute `weston` or `weston-launch`
inside of the weston window, open up a new terminal by clicking on the terminal icon in the top left corner.
In that terminal window write your `waydroid session start` and `waydroid show-full-ui`.

**Comment by bobwatcherx** [comment]
Thanks .and how do i get back to my awesome wm desktop . without restarting
the computer

Pada tanggal Min, 23 Apr 2023 11:34, da HJ ***@***.***>
menulis:

**Comment by HanoJing** [comment]
> Thanks .and how do i get back to my awesome wm desktop . without restarting
the computer

Oh, no you don't have to execute weston from the TTY, you can just run weston from within your awesome WM.
That is the cool part, and when you're done with your work inside of waydroid, just close the weston window.

**Comment by bobwatcherx** [comment]
when I run the Weston-lunch command on the terminal awesome wm an error
appears Weston must be run from a virtual terminal How to handle it.

Pada tanggal Min, 23 Apr 2023 17:14, da HJ ***@***.***>
menulis:

**Comment by bobwatcherx** [comment]
I have problem for open waydroid but not use Weston . And for run Weston i
have error Weston canot run in virtual terminal . How to fix it

Pada tanggal Min, 23 Apr 2023 17:14, da HJ ***@***.***>
menulis:

**Comment by HanoJing** [comment]
> when I run the Weston-lunch command on the terminal awesome wm an error appears Weston must be run from a virtual terminal How to handle it. 

Ah yes, then you must not run `weston-launch` but run just `weston` instead.

**Comment by VenomousSteam81** [comment]
I'm using Fedora with XFCE, and I'm also getting the same issue. I installed mutter instead of weston, found this guide, ran mutter, then my desktop environment had more "life" (it actually had animations). I ended up stopping mutter with Ctrl+c and now my windows are screwed up. Idk what the original compostor is, so I will probably have to have mutter start up automatically. I also installed weston and I will try getting waydroid to work with that.

**Comment by KolFoxy** [comment]
> > how to use in awesome WM iam using Linux MINT os i get error mino@mino-HP-Convertible-x360-11-ab1XX:~$ waydroid session start [10:46:33] WAYLAND_DISPLAY is not set, defaulting to "wayland-0" [10:46:33] Wayland socket '/run/user/1000/wayland-0' doesn't exist; are you running a Wayland compositor?
> 
> If you want to stay on awesome, you can install weston, start weston, open up a terminal inside of weston and start waydroid from within weston.
> 
> `sudo apt install weston` execute `weston` or `weston-launch` inside of the weston window, open up a new terminal by clicking on the terminal icon in the top left corner. In that terminal window write your `waydroid session start` and `waydroid show-full-ui`.

Amazing hack, helped me to get running on Linix Mint with XFCE

**Comment by FrankeyUser** [comment]
i tried with weston but  iget the runtime error: another session is starting the ssion

**Comment by arrahmanbd** [comment]
Change desktop session into wayland
